File: 5-Problem-A-Day/179. Largest Number.java
class Solution {
    public String largestNumber(int[] nums) {
        String[] strs = Arrays.stream(nums).mapToObj(String::valueOf).toArray(String[]::new);
        Arrays.sort(strs, (a, b) -> (b + a).compareTo(a + b));
        return strs[0].equals("0") ? "0" : String.join("", strs);
    }
}

File: 2023/179. Largest Number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        vector<string> ns;
        for (int n : nums) {
            ns.push_back(to_string(n));
        }
        sort(ns.begin(), ns.end(), [](string& left, string& right) {
            return left + right > right + left;
        });
        string res;
        for (int i = 0; i < ns.size(); i++) {
            if (res.empty() && ns[i] == "0") {
                continue;
            }
            res += ns[i];
        }
        return res.empty() ? "0" : res;
    }
};